### ISSUE TITLE
User login enhancements

### ISSUE THREAD

**Issue by zsimpson** [appsoma/ansible-appsoma-mesos#26]
This goes along with the other user stuff you're doing...
Seems like we need the user creation to include pubnlic keys and whether or not that user has sudo access so that we can all login as ourselves.

Z

**Comment by vernonm** [comment]
How much login do we want?  
ssh access to all nodes?  
Ordinary Welder users to have ssh access to the master?  

As far as mesos/docker/nfs are concerned a nologin user on the slaves is just fine, because it's basically `su -u my_welder_user do-stuff.sh` which only needs to resolve to a consistent UID.

The flow I'm worried about is:
* New user signs up for welder
* Creates a password or some other auth, which doesn't matter at all to the actual job, or filesystem
* Somehow we have to create a unique (to the cluster) integer UID and string username for a posix account
* May also need to create a unique group name and GID
* Once that account with a username and uid are created, force an update of /etc/password, /etc/shadow, /etc/group, or insure PAM on nodes can find the user from a service

Basically the problem we got around in V1 with sandbox_* users which, by the way, never had consistent UIDs because we had shared-nothing.

We can add fields like ssh private keys for users, and have group and/or sudoers relationships.

Basically, right now I have the idea of FreeIPA (LDAP/Kerberos packaged) which would support stuff via PAM (might make keys harder), or some combination of shell calls, ansible, nodejs, and a database table.
